src/redisearch_rs/c_entrypoint/types_ffi/src/lib.rs

```diff
@@ -171,7 +171,10 @@ pub unsafe extern "C" fn AggregateResultIter_Next(
 /// - The iterator must have been created using [`AggregateResult_Iter`].
 #[unsafe(no_mangle)]
 pub unsafe extern "C" fn AggregateResultIter_Free(iter: *mut RSAggregateResultIter<'static>) {
-    debug_assert!(!iter.is_null(), "iter must not be null");
+    // Don't free if the pointer is `NULL` - just like the C free function
```

Comment:
Need to update the safety doc above.

Reply:
Done in https://github.com/RediSearch/RediSearch/pull/6536/commits/b7f0fa6c2aa0b03b3478210ad88ee48ce024a177